src/routes/blog/post/nextjs-output-modes/+page.markdoc

```diff
@@ -0,0 +1,98 @@
+
+You might already think, shipping `node_modules` is a very big size hit, and it's true, but not always a bad thing. We will look at why later.
+
+# The "export" mode
+
+Even though we won't be comparing this mode in our benchmarks later in the article, we will still talk about it.
+
+The "export" mode is a mode that allows you to export your app as a static site. This mode is useful if you want to deploy your app as a static site. 
+
+To use this mode, you need to run `next export`. This will create a `out` folder with your static site. You can then serve this folder using any static site hosting provider.
+
+
+# Standalone mode
+
+When building your app in standalone mode, Next.js traces all the files that are required by your app from your `node_modules`. This means that only the functionality your app uses will be bundled into your Next.js build. In this case, you can exclude `node_modules` when calculating sizes, as the build doesn't rely on `node_modules` as it already bundles all necessary code.
+
+This results in bigger `.next/standalone` folder, but we also have to factor in the fact that `node_modules` are essential for the default mode, and not the standalone mode.
+
+The `.next/standalone` folder does not contain any static assets, and Next.js recommends serving them through a CDN. For a minimal setup, you can run a few commands to copy static assets into the standalone directory.
+
+Even with all the benefits, there's a very good reason for this mode not being the default yet.
+
+# Our observations
+
+I created a [benchmark](https://github.com/appwrite-community/nextjs-outputs-bench) to check the sizes of the builds. I had four Next.js apps and ran `next build` on them:
+
+1. Default mode, no dependencies
+2. Standalone mode, no dependencies
+3. Default mode, very heavy dependencies
+4. Standalone mode, very heavy dependencies
+
+Then I compared the sizes of these builds:
+
+```
+━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+  📈 Summary & Comparisons
+━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
+
+  🔹 No Dependencies - Default vs Standalone
+     Default:      431.42 MB (.next + node_modules)
+     Standalone:   57.84 MB
+     📉 Difference:  373.58 MB smaller
+
+  🔹 Heavy Dependencies - Default vs Standalone
+     Default:      712.48 MB (.next + node_modules)
+     Standalone:   57.84 MB
+     📉 Difference:  654.64 MB smaller
+
+  🔹 Impact of Heavy Dependencies
+     Default mode:    +281.06 MB (65.1% increase)
+     Standalone mode: +0 KB (0.0% increase)
+```
+
+The result is clear: standalone mode wins as it's consistently shown lower build sizes. Interestingly, same build size because even though we install heavy dependencies, we aren't actually importing/using them at all in the apps, so these dependencies aren't even getting traced and bundled, which is ideal! Lower build sizes mean faster cold starts on your app.
```

Comment:
```suggestion
The result was clear: standalone mode consistently showed lower build sizes. Interestingly, even when we installed heavy dependencies in the app, we observed that the build size for the apps with standalone mode remained unchanged. Because those dependencies were not being imported and used in the app, they were not bundled, which is ideal. Lower build sizes also result in faster cold starts for your app, which creates a better user experience.
```